src/pybind/ceph_volume_client.py

```diff
@@ -0,0 +1,690 @@
+"""
+Copyright (C) 2015 Red Hat, Inc.
+
+LGPL2.  See file COPYING.
+"""
+
+import json
+import logging
+import os
+import threading
+import errno
+import time
+
+import rados
+import cephfs
+from ceph_argparse import json_command
+
+# Generate missing lib errors at load time, rather than the
+# first time someone tries to use the FS
+try:
+    cephfs.load_libcephfs()
+except EnvironmentError as e:
+    raise ImportError(e.__str__())
+
+
+
+class RadosError(Exception):
+    """
+    Something went wrong talking to Ceph with librados
+    """
+    pass
+
+
+RADOS_TIMEOUT = 10
+SNAP_DIR = ".snap"
+
+log = logging.getLogger(__name__)
+
+
+# Reserved volume group name which we use in paths for volumes
+# that
+NO_GROUP_NAME = "_nogroup"
+
+
+class VolumePath(object):
+    """
+    Identify a volume's path as group->volume
+    The Volume ID is a unique identifier, but this is a much more
+    helpful thing to pass around.
+    """
+    def __init__(self, group_id, volume_id):
+        self.group_id = group_id
+        self.volume_id = volume_id
+        assert self.group_id != NO_GROUP_NAME
+
+    def __str__(self):
+        return "{0}/{1}".format(self.group_id, self.volume_id)
+
```

Comment:
A brief overall description of the module's data model and uses would be good here since this isn't really shared terminology with anything.